### ISSUE TITLE
Potential optimizing training?

### ISSUE THREAD

**Issue by dorpxam** [ZFTurbo/Music-Source-Separation-Training#2]
Hi @ZFTurbo. First of all, thank you very much for this repo, well beyond my expectations (bs-roformer training) with the other models.

### Training settings:
I use the `config_musdb18_mel_band_roformer.yaml` with only two parameters changes: `batch_size: 2` and `target_instrument: drums`.

### Partial training report:
```
------------------------------------------
  Epoch  |   Train loss   |   SDR drums
------------------------------------------
    0    |    2.016814    |    2.3636
    1    |    1.836035    |    2.7670
    2    |    1.768249    |    3.6295
    3    |    1.754987    |    3.6963
    4    |    1.698568    |    3.4976
    5    |    1.665295    |    4.1241
    6    |    1.657550    |    4.4150
    7    |    1.617158    |    4.4690
    8    |    1.600137    |    4.5702
    9    |    1.594573    |    4.6221
   10    |    1.522084    |    4.8008
   11    |    1.524944    |    4.9101
   12    |    1.504670    |    5.0437
   13    |    1.478117    |    5.1029
   14    |    ........    |    ......
------------------------------------------
```
### Training timing:
```
Mean Train Time (1000 steps / epoch) : 4 min 20 sec
Mean Valid Time (  50 steps / epoch) : 8 min 00 sec

EDIT: with a batch_size of 1, I have a better training time : ~ 2 min 30. 
Probably a question of low VRAM size. The valid time stay the same.
```
### Computer settings:
```
CPU : 13th Gen Intel® Core™ i9-13900K
GPU : NVIDIA GeForce RTX 4080 (16GB)
RAM : 64GB
SSD : Kingston FURY Renegade (2TB)
HDD : Seagate SkyHawk AI (16TB)
```
### Questions:
1. What do you think about the training report values (loss/SDR)? Seem good no?
2. Are the timing correct for my configuration?
3. Is the validation duration normal? (2/3 of the total duration of an epoch)
### About the provided pre-trained checkpoints:
The training configuration is set to `1000 epochs` but I would like to know how many epoch you have to do to get the `Vocal SDR 8.42` on the pre-trained (same model)? Currently, for a maximum of 1000 epochs training, the whole duration on my settings will be : 1000 * 12 min 20 sec, so 740000 sec / ~ 12333 min / ~ 205 hours / ~ 8-9 days.
### Potential training time optimization
The current training dataset model is based on direct-from-disk loading for audio chunk. I heard the 'digging' of the reading at each step during training. The MUSDB18HQ is (only) 29GB of size for the whole dataset, so can fit on the RAM on most of modern computer? Don't you think that this could improve the performance of the training steps and that it could be an additional option?

Thank a lot for your answer!

**Comment by ZFTurbo** [comment]
1) With batch_size = 1 you process 1000*1 images per epoch, while with batch_size = 2 you process 2 * 1000 images per epoch so it's where the difference came from

 >> What do you think about the training report values (loss/SDR)? Seem good no?
  For me they are look ok because grow all the time.   
    
 >> Are the timing correct for my configuration?
  Timings look reasonable also
  
 >> Is the validation duration normal? (2/3 of the total duration of an epoch)
I propose you to increase `num_steps` to 5000 or even 10000, so validation will be held not so often.

**Comment by dorpxam** [comment]
> With batch_size = 1 you process 1000*1 images per epoch, while with batch_size = 2 you process 2 * 1000 images per epoch so it's where the difference came from

Yes. When I put the batch_size > 2, I got an ~2 hours estimation for one epoch (and each step are slow), that's why I suppose that I reach the VRAM limit. But yes, I will keep a batch_size = 2.

> I propose you to increase num_steps to 5000 or even 10000, so validation will be held not so often.

OK. I change the value to 5000 (so 10000 estimations by epoch with batch_size 2).

**Comment by ZFTurbo** [comment]
You are right. If you overflow GPU memory then you have great slow down of training.
